packaging/resource_suite_s3_nocache.py

```diff
@@ -2198,6 +2198,73 @@ def test_iput_and_icp_with_invalid_secret_key__issue_6154(self):
                     'DATA_REPL_STATUS', '0'])
             self.user0.run_icommand(['irm', '-f', logical_path])
 
+    def test_s3_in_replication_node__issue_2102(self):
+        resc_prefix = 'issue_2101'
+        repl_resource_name = '%s_repl' % resc_prefix
+        s3_resc1_resource_name = '%s_s3_resc1' % resc_prefix
+        s3_resc2_resource_name = '%s_s3_resc2' % resc_prefix
+        s3_resc3_resource_name = '%s_s3_resc3' % resc_prefix
+        filename = '%s_file' % resc_prefix
+        hostname = lib.get_hostname()
+
+        try:
+
+            # set up resource tree
+            self.admin.assert_icommand("iadmin mkresc %s replication" % repl_resource_name, 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin mkresc %s s3 %s:/%s/%s %s" %
+                                   (s3_resc1_resource_name, hostname, self.s3bucketname, s3_resc1_resource_name, self.s3_context), 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin mkresc %s s3 %s:/%s/%s %s" %
+                                   (s3_resc2_resource_name, hostname, self.s3bucketname, s3_resc2_resource_name, self.s3_context), 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin mkresc %s s3 %s:/%s/%s %s" %
+                                   (s3_resc3_resource_name, hostname, self.s3bucketname, s3_resc3_resource_name, self.s3_context), 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc1_resource_name))
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc2_resource_name))
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc3_resource_name))
+
+            file_sizes = [
+                    100,
+                    4 * 1024 * 1024 - 1,
+                    4 * 1024 * 1024,
+                    8 * 1024 * 1024,
+                    16 * 1024 * 1024,
+                    32 * 1024 * 1024 - 1,
+                    32 * 1024 * 1024,
+                    ];
+
+            for file_size in file_sizes:
+                # create the file
+                lib.make_arbitrary_file(filename, file_size)
+
+                # put the file to repl_resource
+                self.user1.assert_icommand("iput -R %s %s" % (repl_resource_name, filename))
```

Comment:
I made an attempt to do this.  It is in the latest commit:

Some notes:

1. I copied the entire contents of the file into a bytes buf.  If you do multiple istream with append it is really testing something different. 
2. I had to use a local version of make_arbitrary_file updated so that the bytes were all within 0-256.